test/location.test.js

```diff
@@ -13,6 +18,32 @@ define(function(require) {
       window.history.replaceState(null, null, originalPath)
     })
 
+    describe('Instantiation', function() {
+      describe('upgrading from hash path', function() {
+        it('should preserve query parameters', function() {
+          var loc = window.location
+          var originalQuery = new URLSearchParams()
+
+          originalQuery.set('foo', 'bar')
+          originalQuery.set('fizz', 'buzz')
+
+          window.history.replaceState(
+            null, null,
+            loc.origin + loc.pathname +
+            '?' + originalQuery.toString() +
+            '#/app/workspaces'
+          )
+
+          location = getLocationModule(true)
+
+          var newQuery = new URLSearchParams(loc.search.substring(1))
+
+          expect(newQuery.get('foo')).to.equal('bar')
+          expect(newQuery.get('fizz')).to.equal('buzz')
```

Comment:
As I mentioned in my general review, I'm not so sure this case actually makes sense. Any query parameters present in the fragment based path should remain, for sure, but anything in the real path should probably be discarded. Let's discuss that in the more general thread.